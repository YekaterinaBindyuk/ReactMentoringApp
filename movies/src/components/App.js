import React from "react";
import SearchResults from "./search-results/SearchResults";
import ErrorBoundary from "./ErrorBoundary";
import Search from "./search/Search";
import queryString from "query-string";
import actionCreators from "../redux/action-creators/actionCreators";
import { connect } from "react-redux";

export class App extends React.Component {

  render = () => {
    const { searchMovies } = this.props;
    const { searchInput, searchType } = queryString.parse(this.props.location.search);
    if (searchInput && searchType) {
      searchMovies(searchInput, searchType);
    }
    return (
      <div>
        <ErrorBoundary>
          <Search />
          <SearchResults />
        </ErrorBoundary>
      </div>
    );
  };
}

const mapDispatchToProps = {
  searchMovies: actionCreators.searchMovies
};

export default connect(
  null,
  mapDispatchToProps
)(App);
