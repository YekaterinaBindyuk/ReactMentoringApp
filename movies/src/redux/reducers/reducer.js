import * as actions from "../actions/actions";

const initState = {
  movies: [],
  movie: {},
  recommendedMovies: []
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case actions.GET_MOVIE_LIST_SUCCESS:
      return { ...state, movies: action.movies };
    case actions.GET_MOVIE_SUCCESS:
      return { ...state, movie: action.movie };
    case actions.GET_RECOMMENDED_MOVIES_SUCCESS:
      return { ...state, recommendedMovies: action.movies };
    default:
      return state;
  }
}
