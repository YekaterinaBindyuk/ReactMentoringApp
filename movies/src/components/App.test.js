import React from "react";
import { render } from "enzyme";
import App from "./App";
import { Provider } from "react-redux";
import reducer from "../redux/reducers/reducer";
import { createStore } from "redux";

describe("App", () => {
  it('should render correctly in "debug" mode', () => {
    const location = {search: {searchInput: "test", searchType: "title"}};
    const mockStore = createStore(reducer);
    const component = render(
      <Provider store={mockStore}>
        <App debug location={location}/>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
