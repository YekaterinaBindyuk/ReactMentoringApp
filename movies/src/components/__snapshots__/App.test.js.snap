// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<div>
  <ErrorBoundary>
    <AppLogo />
    <Search />
    <MoviesList />
  </ErrorBoundary>
</div>
`;
